products/cms-integration/transformation-process.md

```diff
@@ -0,0 +1,255 @@
+how the sausage is made.
+
+
+<a id="org70bbb4e"></a>
+
+## The actors
+
+The goal of the whole transformer system is to change all the content models
+from one shape to another. In the end, it&rsquo;s just inputs (raw content), process
+(the transformers), and outputs (transformed content).
+
+Before we talk about how it all works, let&rsquo;s get some shared terminology first.
+
+
+<a id="orgd9cac48"></a>
+
+### Content models
+
+A content model has a two-part name, for the purposes of this integration:
+
+1.  A base type
+2.  A sub type
+
+The **base type** is found in the first part of the name of the file in the CMS
+export content directory. For a `node.<UUID>.json` file, its base type is
+`node`. See the section [Raw content](#orgb12cb7b) below for context.
+
+The **sub type** is found *inside* the entity, usually in its `type[0].target_id`
+property.
+
+To get the full name, simply separate them with a hyphen. Examples include
+`node-page` and `paragraph-q_a_section`.
+
+
+<a id="orgb12cb7b"></a>
+
+### Raw content
+
+We start with the raw content as found in the CMS export tarball. When
+extracted, this tarball creates a directory with a child `content/`. In *there*
+are thousands of flat JSON files and a `meta/index.json`, but we can ignore that
+one for now.
+
+Each file is named like `node.<UUID>.json` where `<UUID>` is the UUID of the
+entity.
+
+
+<a id="org07d25fc"></a>
+
+### Transformed content
+
+After content has run through a [transformer](#org3a44eb7), we call it `transformed content`.
+The resulting data structure is what will be used in the [templates](#org187853e).
+
+
+<a id="org3a44eb7"></a>
+
+### Transformer
+
+When you boil down a transformer, all you find is a function that takes an
+object and returns an object. The purpose behind the transformer is to simplify
+using the content in the [templates](#org187853e).
+
+
+<a id="org187853e"></a>
+
+### Liquid templates
+
+The final step before content is translated from an abstract content model to
+full-fledged HTML files, the liquid templates are used to couch the content in
+the necessary trappings of HTML.
+
+For more information on the Liquid template language, see [Shopify&rsquo;s
+documentation](https://shopify.github.io/liquid/).
+
+
+<a id="org48df923"></a>
+
+## The process
+
+Our content&rsquo;s story begins in the gritty metropolis of Metalsmith Pipes, travels
+over the stern Validation Mountains, down into the Vale of the Transformers for
+a life-altering experience, and through the exacting JSON Schema Forest before
+passing through the murky waters of the Liquid Template Lake to come out the
+other side no more a mere collection of abstract content models, but concrete
+HTML files.
+
+
+<a id="org0b02218"></a>
+
+### Metalsmith pipeline
+
+A content build is kicked off by a node script: `yarn build --pull-drupal`
+
+This script is responsible for a lot of things, but for now, the important parts
+are:
+
+1.  Pulling the CMS export tarball
```

Comment:
We won't release this documentation until this is actually implemented, right?

Reply:
Hrm. Good point. I was hoping to release it before it's updated, but yeah. This doesn't describe the current state of the world. :thinking: